src/kernel/init.c

```diff
@@ -620,8 +621,27 @@ void vm_exit(u8 code)
             }
         }
         if (!(shutting_down & SHUTDOWN_POWER)) {
-            if ((code != 0) && get(root, sym(reboot_on_exit)))
-                vm_reset();
+            // reboot_on_exit
+            value reboot = get(root, sym(reboot_on_exit));
+            if (reboot) {
+                u64 reboot_code;
+                if (u64_from_value(reboot, &reboot_code)) {
+                    if (reboot_code == code)
+                        vm_reset();
+                } else if (is_vector(reboot)) {
+                    for (int i = 0; get_u64(reboot, intern_u64(i), &reboot_code); i++) {
+                        if (reboot_code == code)
```

Comment:
Instead of duplicating the code that parses the expected_exit_code option, I would create a `boolean vm_exit_match(u8 exit_code, tuple config, symbol option, boolean match_on_t)` generic function that does the parsing and returns where there is a match (if the option value is "t", it returns the value of `match_on_t`). So vm_exit() can do something like this:
```
if (vm_exit_match(code, root, sym(expected_exit_code), false))
    code = 0;
if (!(shutting_down & SHUTDOWN_POWER)) {
    if (vm_exit_match(code, root, sym(reboot_on_exit), code != 0))
        vm_reset();
```
And yes, I would make vm_exit_match() able to parse a generic `!code` expression where code can be any single value, not just 0.

Reply:
thanks for the advice. Pushed some changes and now all 3 config options ( `expected_exit_code`, `reboot_on_exit` and `idle_on_exit`) benefit from the same enhanced logic (while retaining the existing default behavior).

While matching vector is supported ["0","1","6"], I did not implement negating vector ["!0", "!1", "!6"]. Do we want it?